syntax = "proto3";

package silenteight.datasource.api.country;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.country";
option java_outer_classname = "CountryInputProto";
option java_multiple_files = true;

service CountryInputService {
  rpc BatchGetMatchCountryInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchCountryInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/country/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchCountryInputsV1Response {
  repeated CountryInput country_matches = 1;
}

message CountryInput {
  string match = 1;
  repeated CountryFeatureInput country_feature_inputs = 2;
}

message CountryFeatureInput {
  string feature = 1;
  repeated string alerted_party_countries = 2;
  repeated string watchlist_countries = 3;
}
